src/Core/Managed/Shared/Extensibility/TelemetryConfiguration.cs

```diff
@@ -313,6 +314,43 @@ public void Dispose()
         }
 
         /// <summary>
+        /// Adds a new telemetry sink to the configuration.
+        /// </summary>
+        /// <param name="sink">Sink to add.</param>
+        public void AddSink(TelemetrySink sink)
```

Comment:
I'd than have a default sink and sink collection. Exposing collection is the way SDK configures things via config file